FTB/Signatures/tests.py

```diff
@@ -874,528 +874,782 @@ def runTest(self):
-        self.assertEqual(crashInfo.backtrace[8], "js::jit::InvokeFunction")
-        self.assertEqual(crashInfo.backtrace[9], "??")
-        self.assertEqual(crashInfo.backtrace[10], "EnterIon")
-        self.assertEqual(crashInfo.backtrace[11], "js::jit::IonCannon")
-        self.assertEqual(crashInfo.backtrace[12], "Interpret")
-        self.assertEqual(crashInfo.backtrace[13], "js::RunScript")
-        self.assertEqual(crashInfo.backtrace[14], "js::ExecuteKernel")
-        self.assertEqual(crashInfo.backtrace[15], "js::Execute")
-        self.assertEqual(crashInfo.backtrace[16], "Evaluate")
-        self.assertEqual(crashInfo.backtrace[17], "Evaluate")
-        self.assertEqual(crashInfo.backtrace[18], "JS::Evaluate")
-        self.assertEqual(crashInfo.backtrace[19], "EvalInContext")
-        self.assertEqual(crashInfo.backtrace[20], "??")
-        self.assertEqual(crashInfo.backtrace[21], "Print")
-        self.assertEqual(crashInfo.backtrace[22], "??")
-        self.assertEqual(crashInfo.backtrace[23], "??")
-        self.assertEqual(crashInfo.backtrace[24], "??")
-        self.assertEqual(crashInfo.backtrace[25], "??")
-        self.assertEqual(crashInfo.backtrace[26], "??")
-        self.assertEqual(crashInfo.backtrace[27], "??")
-        self.assertEqual(crashInfo.backtrace[28], "js::Activation::Activation")
-        self.assertEqual(crashInfo.backtrace[29], "EnterIon")
-        self.assertEqual(crashInfo.backtrace[30], "js::jit::IonCannon")
-        self.assertEqual(crashInfo.backtrace[31], "js::RunScript")
-        self.assertEqual(crashInfo.backtrace[32], "js::Invoke")
-        self.assertEqual(crashInfo.backtrace[33], "js::Invoke")
-        self.assertEqual(crashInfo.backtrace[34], "js::jit::DoCallFallback")
-        self.assertEqual(crashInfo.backtrace[35], "??")
-        self.assertEqual(crashInfo.backtrace[36], "EnterBaseline")
-        self.assertEqual(crashInfo.backtrace[37], "js::jit::EnterBaselineMethod")
-        self.assertEqual(crashInfo.backtrace[38], "js::RunScript")
-        self.assertEqual(crashInfo.backtrace[39], "js::ExecuteKernel")
-        self.assertEqual(crashInfo.backtrace[40], "js::Execute")
-        self.assertEqual(crashInfo.backtrace[41], "ExecuteScript")
-        self.assertEqual(crashInfo.backtrace[42], "JS_ExecuteScript")
-        self.assertEqual(crashInfo.backtrace[43], "RunFile")
-        self.assertEqual(crashInfo.backtrace[44], "ProcessArgs")
-        self.assertEqual(crashInfo.backtrace[45], "Shell")
-        self.assertEqual(crashInfo.backtrace[46], "main")
-        self.assertEqual(crashInfo.backtrace[47], "__tmainCRTStartup")
-        self.assertEqual(crashInfo.backtrace[48], "mainCRTStartup")
-        self.assertEqual(crashInfo.backtrace[49], "BaseThreadInitThunk")
-        self.assertEqual(crashInfo.backtrace[50], "RtlInitializeExceptionChain")
-        self.assertEqual(crashInfo.backtrace[51], "RtlInitializeExceptionChain")
+        self.assertEqual(len(crashInfo.backtrace), 33)
```

Comment:
With such long backtraces, you don't need to check every single frame. When writing tests, you should always make sure to cover edge cases and maybe a few generic cases. E.g. you always want to check backtrace[0], you always want to check the last stackframe as well (bounds) and maybe 2-3 random frames in the middle. If there are frames with e.g. special properties or chars, that you think might be troublesome, include them. But you don't need to check them all :) You can still leave this as it is of course.